### ISSUE TITLE
Possible name clash of relations when generating source files with "-g" 

### ISSUE THREAD

**Issue by limasierra** [souffle-lang/souffle#226]
Hi,

I ran into the following problem today:

```
.comp MyComponent {
	.decl value(X:number)
}

.init boolean = MyComponent

.decl boolean_value(X:number) output

boolean_value(X) :- boolean.value(X).

boolean.value(1).
```

When generating source code with `souffle -g analysis.cpp analysis.dl`, the compilation ( `g++ -I/path/to/souffle/include -lsqlite3 analysis.cpp` ) will fail with errors like the following:

```
analysis.cpp:19:18: error: redeclaration of ‘souffle::ram::Relation<1u> souffle::Sf_analysis::rel_boolean_value’
 ram::Relation<1> rel_boolean_value;
                  ^~~~~~~~~~~~~~~~~
analysis.cpp:17:18: note: previous declaration ‘souffle::ram::Relation<1u> souffle::Sf_analysis::rel_boolean_value’
 ram::Relation<1> rel_boolean_value;
                  ^~~~~~~~~~~~~~~~~
```

Obviously, the name-mangling produces conflicting names.

Apart from a "clean fix" where different relations will _always_ have different names, I personally think the name-mangler should at least use multiple underscores for now.

Again, note that a "normal" run with `souffle -D- analysis.dl` works fine!

Kind regards,
Laurent

**Comment by b-scholz** [comment]
Good  point. We had similar issues in the past with special characters; we need to extend the name mapping for theses cases as well.

**Comment by lyndonhenry** [comment]
This has been closed as of #268.
